src/session-user/session-user.controller.ts

```diff
@@ -1,4 +1,38 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Param, Post, Req, UseGuards } from '@nestjs/common';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { Request } from 'express';
+import { FirebaseAuthGuard } from '../firebase/firebase-auth.guard';
+import { IFirebaseUser } from '../firebase/firebase-user.interface';
+import { ControllerDecorator } from '../utils/controller.decorator';
+import { CreateSessionUserDto } from './dtos/create-session-user.dto';
+import { SessionUserService } from './session-user.service';
 
-@Controller('session-user')
-export class SessionUserController {}
+@ApiTags('Session User')
+@ControllerDecorator()
+@Controller('/v1/session-user')
+export class SessionUserController {
+  constructor(private readonly sessionUserService: SessionUserService) {}
+
+  @Post()
+  @ApiBearerAuth()
+  @UseGuards(FirebaseAuthGuard)
+  async createSessionUserRecord(
+    @Req() req: Request,
+    @Body() createSessionUserDto: CreateSessionUserDto,
+  ) {
+    return await this.sessionUserService.createSessionUser(
+      req['user'] as IFirebaseUser,
+      createSessionUserDto,
+    );
+  }
+
+  @Post(':sessionId')
+  @ApiBearerAuth()
+  @UseGuards(FirebaseAuthGuard)
+  async updateSessionUserRecord(@Req() req: Request, @Param() params) {
+    return await this.sessionUserService.updateSessionUser(
+      req['user'] as IFirebaseUser,
+      params.sessionId,
+    );
+  }
+}
```

Comment:
minor naming (again, sorry 😝) 
here we're using `[create/update]SessionUserRecord` but elsewhere we're just using  `[create/update]PartnerAccess`

i think lose the 'record' from the functions here